### ISSUE TITLE
Extend core templates

### ISSUE THREAD

**Issue by mikhailshilkov** [Azure/azure-functions-cli#305]
The selection of templates for `core` branch is quite limited.

I would expect at least
- F# templates on par with C#
- Precompiled functions templates

I can see that templates are loaded from `https://functionscdn.azureedge.net/public/templates.json`. 
- Is there a way to send a pull request to this file?
- Is there a way to substitute this file with my own (for personal use / demos)?

**Comment by fabiocav** [comment]
Yeah, we hope to have some cycles to focus on the F# experience soon.

The templates are kept here: https://github.com/azure/azure-webjobs-sdk-templates

With some Fiddler filters, you can set a response for that API and return your own payload (either from a local file or something on GitHub, for example).

**Comment by mikhailshilkov** [comment]
Thanks for the pointers @fabiocav 

If I understand it correctly, CLI uses portal templates. Does it mean there are no plans to add precompiled templates to CLI (since they won't work in the portal)?

**Comment by PoisonousJohn** [comment]
@mikhailshilkov precompiled functions can be deployed from VS, so they are supposed to be working in portal. Thus you can write a template (for that repository: https://github.com/azure/azure-webjobs-sdk-templates).

I also need that template, since precompiled functions should be much faster.

**Comment by mikhailshilkov** [comment]
@PoisonousJohn Yes, "won't work" was bad wording. I meant that there isn't much sense to create precompiled functions in the portal directly, since they will be read-only.

**Comment by PoisonousJohn** [comment]
@fabiocav does this CLI use "VS" templates? I want to understand what kind of template should I add to support precompiled .net core project.

**Comment by nmfisher** [comment]
Bumping this - any ETA on F# templates for v2?

This is my first time dealing with Azure Functions, and I must say it's a tad confusing. Most of the blog posts were written about 1.0 (where F# templates were available for .NET Framework on Windows), but didn't explicitly refer to the version number. I was unaware of this, so created a project using 2.0 (which has no templates), and had no idea why "func new" wasn't allowing me to select "F#".

**Comment by Ma3yTa** [comment]
@fabiocav where we can found templates for v2?
How we can make changes to https://functionscdn.azureedge.net/public/templates.json ?
Thanks.

**Comment by Ma3yTa** [comment]
@fabiocav any news?
